src/gpu/eigensolver.cpp

```diff
@@ -102,7 +123,7 @@ auto eigh(ContextInternal& ctx, T wl, ConstDeviceView<api::ComplexType<T>, 2> s,
     auto gReduced = queue.create_device_array<api::ComplexType<T>, 2>({nBeamReduced, nBeamReduced});
     gram_matrix<T>(ctx, wReduced, xyz, wl, gReduced);
 
-    eigensolver::solve(queue, mode, 'L', nBeam, v.data(), v.strides(1), gReduced.data(),
+    eigensolver::solve(queue, mode, 'L', nBeamReduced, v.data(), v.strides(1), gReduced.data(),
```

Comment:
Nice catch. Did this fix the eigensolver issues you saw?

Reply:
Yes, that was the issue when rows/columns where flagged.